From f68a942c1cf1f904188bca1854edf37445182b0d Mon Sep 17 00:00:00 2001
From: Liu Ying <victor.liu@nxp.com>
Date: Wed, 13 Sep 2017 16:53:58 +0800
Subject: [PATCH 4522/5242] MLK-19413-11 gpu: imx: dpu: common: Add di_grp_id
 in display client pdev's data

commit  56ebf3944ad61e1e84fe128d3def9e242099dbcd from
https://source.codeaurora.org/external/imx/linux-imx.git

This patch adds a new di_grp_id entry in display client pdev's data
so that the relevant display platform driver may know the display
group ID of the display device.

Signed-off-by: Liu Ying <victor.liu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/imx/dpu/dpu-common.c |    4 +++-
 include/video/dpu.h              |    1 +
 2 files changed, 4 insertions(+), 1 deletion(-)

diff --git a/drivers/gpu/imx/dpu/dpu-common.c b/drivers/gpu/imx/dpu/dpu-common.c
index 4e31c1e..ef7380a 100644
--- a/drivers/gpu/imx/dpu/dpu-common.c
+++ b/drivers/gpu/imx/dpu/dpu-common.c
@@ -1188,8 +1188,10 @@ static int dpu_add_client_devices(struct dpu_soc *dpu)
 			}
 		}
 
-		if (is_disp)
+		if (is_disp) {
 			reg[i].pdata.plane_grp = plane_grp;
+			reg[i].pdata.di_grp_id = plane_grp->id;
+		}
 
 		pdev = platform_device_alloc(reg[i].name, id++);
 		if (!pdev) {
diff --git a/include/video/dpu.h b/include/video/dpu.h
index bf31e0d..307f496 100644
--- a/include/video/dpu.h
+++ b/include/video/dpu.h
@@ -799,6 +799,7 @@ static inline struct dpu_plane_grp *plane_res_to_grp(struct dpu_plane_res *res)
 
 struct dpu_client_platformdata {
 	const unsigned int	stream_id;
+	unsigned int		di_grp_id;
 	struct dpu_plane_grp	*plane_grp;
 
 	struct device_node	*of_node;
-- 
1.7.9.5

